spec.bs

```diff
@@ -820,20 +877,32 @@ A <dfn export for=fencedframetype>fenced frame reporter</dfn> is a [=struct=] wi
 
 <div algorithm>
   In order to <dfn>report an event</dfn> using a [=fencedframetype/fenced frame reporter=]
-  |reporter| with a {{FenceReportingDestination}} |destination|, [=string=] |eventType|, and
-  [=string=] |eventData|, run these steps:
+  |reporter| with a {{FenceReportingDestination}} |destination|, and a
+  [=fencedframetype/destination event=] |event|, run these steps:
+
+  1. Let |metadata| be |reporter|'s
+     [=fenced frame reporter/fenced frame reporting metadata reference=].
 
   1. If |destination| is `"direct-seller"`:
 
-     1. If |reporter|'s [=fenced frame reporter/fenced frame reporting metadata reference=]'s
-        [=fenced frame reporting metadata/direct seller is seller=] is true, set |destination| to
-        `"seller"`.
+     1. If |metadata|'s [=fenced frame reporting metadata/direct seller is seller=] is true, set
+        |destination| to `"seller"`.
 
      1. Otherwise, set |destination| to `"component-seller"`.
 
-  1. Let |reporting map| be a reference to |reporter|'s [=fenced frame reporter/fenced frame
-     reporting metadata reference=]'s [=fenced frame reporting metadata/fenced frame reporting
-     map=].
+  1. If |event| is a [=fencedframetype/destination URL event=]:
+     1. If |event|'s [=url/origin=] is not [=same origin=] with any of the entries in
```

Comment:
IIRC, the implementation also checks if the event's origin is enrolled. Are we assuming that this same-origin check with `allowed reporting origins` covers that? If so, would be good to document at [line 750](https://github.com/WICG/fenced-frame/pull/113/files#diff-6f5a1d8263b0b0c42e2716ba5750e3652e359532647ac934c1c70086ae3ceddaR750) that all origins in [=fenced frame reporting metadata/allowed reporting origins=] are enrolled.

Reply:
Attestation checks are performed when sending the beacon, i.e. it uses an already present check on line 823.

Reply:
Ack. Forget about the "interesting" `optionally` part.